plugin/storage/es/spanstore/reader.go

```diff
@@ -293,7 +293,7 @@ func (s *SpanReader) multiRead(ctx context.Context, traceIDs []string, startTime
 		if len(traceIDs) == 0 {
 			break
 		}
-
+		searchRequests := make([]*elastic.SearchRequest, len(traceIDs))
```

Comment:
this isn't the most efficient fix, ideally we alloc just once and play around with the indices rather than alloc each iteration. We can leave a TODO for now.

Reply:
Good suggestion and we(@crazycomputer) have fixed it .We will test it in our project tommorrow and if success,i will make a standard PR again.

Reply:
> this isn't the most efficient fix, ideally we alloc just once and play around with the indices rather than alloc each iteration. We can leave a TODO for now.

Good suggestion and we(@crazycomputer) have fixed it .We will test it in our project tommorrow and if success,i will make a standard PR again.

Reply:
> this isn't the most efficient fix, ideally we alloc just once and play around with the indices rather than alloc each iteration. We can leave a TODO for now.

Good suggestion.We(@crazycomputer) have fixed it and will test it in our project tommorrow.If success,i will make a standard PR again.